server/src/main/java/org/opensearch/common/geo/PointParser.java

```diff
@@ -0,0 +1,227 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.common.geo;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.geometry.Point;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Locale;
+
+/**
+ * A utility class with a point parser methods supporting different point representation formats
+ *
+ * @opensearch.internal
+ */
+public class PointParser {
+    private static final String FN_GEOJSON_TYPE = "type";
+    private static final String FV_GEOJSON_TYPE_POINT = "Point";
+    private static final String FN_GEOJSON_COORDS = "coordinates";
+
+    private static final String FN_GEOHASH = "geohash";
+    private static final String ERR_MSG_INVALID_TOKEN = "token [{}] not allowed";
+
+    private String fieldNameX;
+    private String fieldNameY;
+    private boolean supportGeoHash;
+
+    private String invalidFieldErrMsg;
+
+    public PointParser(final String fieldNameX, final String fieldNameY, final boolean supportGeoHash) {
+        this.fieldNameX = fieldNameX;
+        this.fieldNameY = fieldNameY;
+        this.supportGeoHash = supportGeoHash;
+        this.invalidFieldErrMsg = supportGeoHash
+            ? String.format(
+                Locale.ROOT,
+                "field must be either %s/%s, %s/%s, or %s",
+                fieldNameX,
+                fieldNameY,
+                FN_GEOJSON_TYPE,
+                FN_GEOJSON_COORDS,
+                FN_GEOHASH
+            )
+            : String.format(
+                Locale.ROOT,
+                "field must be either %s/%s, or %s/%s",
+                fieldNameX,
+                fieldNameY,
+                FN_GEOJSON_TYPE,
+                FN_GEOJSON_COORDS
+            );
+    }
+
+    public Point parseObject(final XContentParser parser, final boolean ignoreZValue, final GeoUtils.EffectivePoint effectivePoint)
```

Comment:
why functions of this class are not static?

Reply:
In general, static method should be minimized in OOP design. I am surprised how many static methods are there in OpenSearch repo. Static method is harder to test than non-static method also.

Reply:
Yes I understand it is harder to test, but if a method is not storing the state and doing some computation its better to make it static as it will avoid people mocking that function. I would recommend lets make it static so we don't need to maintain/create the objects for that class again.